modules/islandora_binary_object_zip_importer/includes/importer.inc

```diff
@@ -185,6 +185,7 @@ class BinaryObjectZipBatchImportObject extends ZipBatchImportObject {
       $zip->open(drupal_realpath($this->source['file']->uri));
       if ($stream = $zip->getStream($name)) {
         $path = $default_scheme . basename($name);
+        file_unmanaged_save_data($stream, $path);
```

Comment:
The return of this should be captured and used in the `islandora_temp_file_entry()` call below, since the name with which the data _actually_ gets saved may be different (due to the `FILE_EXISTS_RENAME` business, which can add a numerical suffix). Something like:
```php
 if ($stream = $zip->getStream($name)) {
  $uri = file_unmanaged_save_data($stream, $default_scheme . basename($name));
  $file = islandora_temp_file_entry($uri, $mimetype);
  [... the rest ...]
```